### PR TITLE
closes #178: findOne and findOneUpdate per spec, improved tests

### LINKED ISSUE #178: Confirm all response messages against spec
Issue #175 has updates to the API spec that locks in the response messages from a meeting today.

The API spec now has a general description of the request and response, and each command specifies the data is should return. 

This is a catchall ticket to ensure those changes to the spec and the code are in sync. There may be some  changes that have already happened like removing the updatedId's list from the status. 

I think the spec changes are good based on the meeting today, and we can probably start implementing. 

@maheshrajamani  and @ivansenic  this also includes the response data when there are errors. Still have changes for #160 in  my git stash, but this describes what the response should be.

### PR DESCRIPTION
**Pull request by ivansenic**
**What this PR does**:

Adaptation to spec for the `findOne` & `findOneAndUpdate`, including test improvements.

* [x] removed option class from `findOne` command as per spec no options exists
* [x] validation and documentation of options in the `findOneAndUpdate`
* improved tests:
   * [x] added command validation tests
   * [x] refactored, extended resolver tests, test all options 
   * [x] refactored operation tests
   * [x] added error case in operation tests
   * [x] several fixes for the integration tests:
      * added cases with not matching and missing fields as well
      * assert no errors & data
      * improved readability
      * remove test order - not needed
* [x] fixed swagger examples

**Which issue(s) this PR fixes**:
Closes to #178. (final task)

### CODE REVIEW

**Comment by ivansenic** [comment]
@vkarpov15 I need your help quickly. I specified in the `updateOne`, `updateMany` and `findOneAndUpdate` that the `update` clause is mandatory. Does this make any sense or should I revert?

I was thinking there is no sense in not having the `update` clause, but since there is `upsert` option, commands like this do make sense for example:

```
{
  "findOneAndUpdate": {
    "filter" : {"key": "value"},
    "options": {
       "upsert": true
    }
  }
}
``` 

I think I should revert and make `update` optional?

File: src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java [hunk 1]
```diff
@@ -17,15 +18,26 @@
...
               message = "returnDocument value can only be 'before' or 'after'")
+          @Schema(
+              description =
+                  "Specifies which document to perform the projection on. If `before` the projection is performed on the document before the update is applied, if `after` the document projection is from the document after the update.",
+              defaultValue = "before")
           String returnDocument,
-      boolean upsert) {}
+      @Schema(
+              description =
+                  "When `true`, if no documents match the `filter` clause the command will create a new _empty_ document and apply the `update` clause to the empty document.",
```
Comment (tatu-at-datastax) [inline src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java:40]: Strictly speaking it's not exactly empty I think, as some values are initialized from filter clause.

@maheshrajamani Is this correct? And if so, how are default values from filter clause selected/applied (only equality ones)

Comment (maheshrajamani) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java:40]: Yes, When new document is created all equals filter's are added to the document.

Comment (ivansenic) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java:40]: Will fix. Should be fixed in update commands as well

Comment (vkarpov15) [reply src/main/java/io/stargate/sgv2/jsonapi/api/model/command/impl/FindOneAndUpdateCommand.java:40]: @maheshrajamani is right about this one, any equals filters are added to the document before updates are applied.

**Review by vkarpov15 — APPROVED** [review]
One quick comment but otherwise LGTM

**Comment by ivansenic** [comment]
> @vkarpov15 I need your help quickly. I specified in the `updateOne`, `updateMany` and `findOneAndUpdate` that the `update` clause is mandatory. Does this make any sense or should I revert?
> 
> I was thinking there is no sense in not having the `update` clause, but since there is `upsert` option, commands like this do make sense for example:
> 
> ```
> {
>   "findOneAndUpdate": {
>     "filter" : {"key": "value"},
>     "options": {
>        "upsert": true
>     }
>   }
> }
> ```
> 
> I think I should revert and make `update` optional?

@vkarpov15 about about this one? :arrow_up:

**Review by maheshrajamani — APPROVED** [review]
LGTM

**Pull request merged**